packages/payment-detection/src/any/any-to-erc20-proxy-info-retriever.ts

```diff
@@ -0,0 +1,160 @@
+   * @param paymentReference The reference to identify the payment
+   * @param conversionProxyContractAddress The address of the proxy contract
+   * @param conversionProxyCreationBlockNumber The block that created the proxy contract
+   * @param toAddress Address of the balance we want to check
+   * @param eventName Indicate if it is an address for payment or refund
+   * @param network The Ethereum network to use
+   */
+  constructor(
+    private requestCurrency: RequestLogicTypes.ICurrency,
+    private paymentReference: string,
+    private conversionProxyContractAddress: string,
+    private conversionProxyCreationBlockNumber: number,
+    private erc20FeeProxyContractAddress: string,
+    private erc20FeeProxyCreationBlockNumber: number,
+    private toAddress: string,
+    private eventName: PaymentTypes.EVENTS_NAMES,
+    private network: string,
+    private acceptedTokens?: string[],
+    private maxRateTimespan: number = 0,
+  ) {
+    // Creates a local or default provider
+    this.provider =
+      this.network === 'private'
+        ? new ethers.providers.JsonRpcProvider()
+        : ethers.getDefaultProvider(this.network);
+
+    // Setup the conversion proxy contract interface
+    this.contractConversionProxy = new ethers.Contract(
+      this.conversionProxyContractAddress,
+      erc20ConversionProxyContractAbiFragment,
+      this.provider,
+    );
+
+    this.contractERC20FeeProxy = new ethers.Contract(
+      this.erc20FeeProxyContractAddress,
+      erc20FeeProxyContractAbiFragment,
+      this.provider,
+    );
+
+    this.acceptedTokens = acceptedTokens?.map((token) => token.toLowerCase());
+  }
+
+  /**
+   * Retrieves transfer events for the current contract, address and network.
+   */
+  public async getTransferEvents(): Promise<PaymentTypes.ERC20PaymentNetworkEvent[]> {
+    // Create a filter to find all the Fee Transfer logs with the payment reference
+    const conversionFilter = this.contractConversionProxy.filters.TransferWithConversionAndReference(
+      null,
+      null,
+      '0x' + this.paymentReference,
+    ) as ethers.providers.Filter;
+    conversionFilter.fromBlock = this.conversionProxyCreationBlockNumber;
+    conversionFilter.toBlock = 'latest';
+
+    // Get the fee proxy contract event logs
+    const conversionLogs = await this.provider.getLogs(conversionFilter);
+
+    // Create a filter to find all the Fee Transfer logs with the payment reference
+    const feeFilter = this.contractERC20FeeProxy.filters.TransferWithReferenceAndFee(
+      null,
+      null,
+      null,
+      '0x' + this.paymentReference,
+      null,
+      null,
+    ) as ethers.providers.Filter;
+    feeFilter.fromBlock = this.erc20FeeProxyCreationBlockNumber;
+    feeFilter.toBlock = 'latest';
+
+    // Get the fee proxy contract event logs
+    const feeLogs = await this.provider.getLogs(feeFilter);
+
+    // Parses, filters and creates the events from the logs with the payment reference
```

Comment:
I think it's worth explaining a bit more why we parse 2 sets of logs and how 2 events relate to each other.

Reply:
The link between the two events is explained line 68. Is it enough?